src/tracer.cpp

```diff
@@ -126,7 +126,8 @@ std::unique_ptr<ot::Span> Tracer::StartSpanWithOptions(ot::string_view operation
 
   // Check early if we need to discard. Check at span Finish if we need to sample (since users can
   // set this).
-  if (sampler_->discard(span_context)) {
+  if (span_context.getPropagatedSamplingPriority() == nullptr && sampler_->discard(span_context)) {
```

Comment:
Would it make sense to move this check `span_context.getPropagatedSamplingPriority() == nullptr` to `DiscardRateSampler` ?

Reply:
That could also work.
To me that detail is slightly more of a `tracer` decision than the `sampler` decision, and I'd rather not bury it.

Reply:
Makes sense 👍